doc/TYPES2021/Demos/HSP.tex

```diff
@@ -2119,30 +2010,27 @@ \subsection{Algebras}\label{algebras}
 \>[0]\<%
 \end{code}
 \fi
-\noindent To see why the \af{Lift-Alg} function is useful, recall that our definition of the algebra
-record type uses two universe level parameters corresponding to those of the algebra's
-underlying domain setoid.
+\noindent Recall that our definition of algebra uses two universe levels, corresponding
+to those of the domain setoid.
 \ifshort\else
 Concretely, an algebra of type \ar{Algebra} \ab{α} \ab{ρᵃ} has a
 \afld{Domain} of type \ar{Setoid} \ab{α} \ab{ρᵃ}. This packages a ``carrier set''
 (\afld{Carrier}), inhabiting \ap{Type} \ab{α}, with an equality on \afld{Carrier} of type
 \af{Rel} \afld{Carrier} \ab{ρᵃ}.
 \fi
-The \af{Lift-Alg} function takes an algebra---one whose carrier inhabits \ap{Type \ab{α}}
-with equality of type \af{Rel} \afld{Carrier} \ab{ρᵃ}---and constructs a new algebra whose
+\af{Lift-Alg} takes an algebra parametrized by levels \ab{a} and \ab{ρᵃ}
+and constructs a new algebra whose
 carrier inhabits \ap{Type} (\ab{α} \ap{⊔} \ab{ℓ₀}) with equality of type \af{Rel}
-\afld{Carrier} (\ab{ρᵃ} \ap{⊔} \ab{ℓ₁}). This lifting operation would be worthless without
-a useful semantic connection between the input and output algebras.
-Fortunately, it is easy to prove that \af{Lift-Alg} is an \defn{algebraic invariant},
-which is to say that the resulting ``lifted'' algebra has the same algebraic properties as
-the original algebra, a fact we will codify later in a type called \af{Lift-≅}.
+\afld{Carrier} (\ab{ρᵃ} \ap{⊔} \ab{ℓ₁}). To be useful, the resulting algebra should
+have the same semantic properties as the input algebra, which is indeed the case,
+as we will show later.
 
 \paragraph*{Product Algebras}
-Here we review the (informal) definition of the \defn{product} of a family of
-\ab{𝑆}-algebras and then define a type which formalizes this notion in \agda.
+Let us first give the (informal) definition of the \defn{product} of a family of
```

Comment:
"Let us first give..."

First?  And then what?  (You're leaving the reader hanging.)

(I prefer the original wording.)

Reply:
Sure for the "Here we review".  But less so for "a type which formalizes this notion in \agda".  The whole paper is about formalizing in Agda, it is tiresome to keep saying that.

Reply:
Haha, yes, very good point!